lib/compiler/src/beam_validator.erl

```diff
@@ -995,6 +995,18 @@ vi(raw_raise=I, Vst0) ->
 
 vi(bs_init_writable=I, Vst) ->
     validate_body_call(I, 1, Vst);
+vi({bs_create_bin,{f,Fail},Heap,Live,Unit,Dst,{list,List}}, Vst0) ->
+    verify_live(Live, Vst0),
+    verify_y_init(Vst0),
+    verify_create_bin_list(List, Vst0),
+    Vst = heap_alloc(Heap, Vst0),
+    branch(Fail, Vst,
+           fun(SuccVst0) ->
+                   SuccVst1 = update_create_bin_list(List, SuccVst0),
+                   SuccVst = prune_x_regs(Live, SuccVst1),
```

Comment:
If I'm reading the instruction right we can fail after a garbage collection, so we should prune X registers unconditionally.